monasca_events_api/tests/test_transforms.py

```diff
@@ -0,0 +1,347 @@
+    def test_on_get_fail_db_down(
+            self,
+            helper_tenant_id,
+            helpers_validate,
+            mysqlRepo):
+        """GET Method fail due to db down"""
+        mysqlRepo.connect.side_effect = repository_exceptions.RepositoryException(
+            "Database Connection Error")
+        helpers_validate.return_value = True
+        mysqlRepo.connect.return_value = True
+        helper_tenant_id.return_value = '0ab1ac0a-2867-402d'
+
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            transObj.on_get(self._generate_req(), res)
+            self.assertFalse(
+                1,
+                msg="Database Down,  GET should fail but succeeded")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPInternalServerError)
+            self.assertEqual(e.status, '500 Internal Server Error')
+
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    def test_on_get_fail_validate_authorization(self, _validate_authorization):
+        """GET Method fail due to validate authorization"""
+        _validate_authorization.side_effect = falcon.HTTPUnauthorized(
+            'Forbidden',
+            'Tenant does not have any roles')
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            transObj.on_get(self._generate_req(), res)
+        except Exception as e:
+            self.assertRaises(falcon.HTTPUnauthorized)
+            self.assertEqual(e.status, '401 Unauthorized')
+
+    @mock.patch('monasca_events_api.v2.transforms.Transforms._list_transforms')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_get_pass(
+            self,
+            helper_tenant_id,
+            helpers_validate,
+            list_transforms):
+        """GET Method success Single Event"""
+        helpers_validate.return_value = True
+        helper_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        returnStreams = [{"region": "useast", "tenantId": "0ab1ac0a-2867-402d",
+                          "creation_time": "1434331190", "transform": "1"},
+                         {"region": "useast", "tenantId": "0ab1ac0a-2866-403d",
+                          "creation_time": "1234567890", "transform": "2"}]
+        list_transforms.return_value = returnStreams
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        transObj.on_get(self._generate_req(), res)
+        self.assertEqual(res.status, '200 OK')
```

Comment:
Do you have an on_get test to test getting with an ID?  Also you should test for the standard pagination fields when getting everything in the DB.  Those fields won't exist yet but you can add them along with the test.  Take a look at the metrics resource in the monasca_api to see how it's done.